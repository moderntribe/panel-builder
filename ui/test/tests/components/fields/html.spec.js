/* global expect b:true */

import React from 'react';
import { shallow, mount } from 'enzyme';
import Html from 'components/fields/html';
import renderer from 'react-test-renderer';

describe('HTML field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Html />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<Html />);
		expect(wrapper.type()).toEqual('div');
	});

	it('allows us to set props', () => {
		const wrapper = mount(<Html description="<p>Some HTML text <b>goes</b> <i>here</i></p>" />);
		expect(wrapper.props().description).toEqual('<p>Some HTML text <b>goes</b> <i>here</i></p>');
		wrapper.setProps({ description: 'Change props for test' });
		expect(wrapper.props().description).toEqual('Change props for test');
	});
});
