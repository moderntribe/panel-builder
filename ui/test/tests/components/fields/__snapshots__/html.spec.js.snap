exports[`HTML field renders correctly 1`] = `
<div
  className="field panel-field">
  <div
    className="panel-field-description"
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    } />
</div>
`;
